src/lib/EventListener/InteractiveLoginListener.php

```diff
@@ -37,8 +63,78 @@ public static function getSubscribedEvents()
+            $uidKey         = $this->config['ldap']['search']['uid_key'];
+            $passwordAttr   = $this->config['ldap']['search']['password_attribute'];
+            $targetGroup    = $this->config['ez_user']['target_usergroup'];
+            $emailAttr      = $this->config['ez_user']['email_attr'];
+            $attributes     = $this->config['ez_user']['attributes'];
+
+            // Login to LDAP server and get user attributes
+            $this->ldap->bind($searchDn, $password);
+            $queryString = str_replace("{username}", $username, $queryString);
+            $queryString = str_replace("{uid_key}", $uidKey, $queryString);
+            $query = $this->ldap->query($baseDn, $queryString);
+            $results = $query->execute();
+
+            // Prepare user details
+            $propertyAccessor = PropertyAccess::createPropertyAccessor();
+
+            $ldapUser   = $results->toArray()[0]->getAttributes();
+            $email      = $propertyAccessor->getValue($ldapUser, "[$emailAttr][0]");
+            $password   = $propertyAccessor->getValue($ldapUser, "[$passwordAttr][0]");
+
+            $user = $userService->newUserCreateStruct($username, $email, $password, "fre-FR");
+
+            foreach ($attributes as $attr) {
+                $value = $propertyAccessor->getValue($ldapUser, "[{$attr['ldap_attr']}][0]");
+                if ($value) {
+                    $user->setField($attr['user_attr'], $value);
+                } else {
+                    // TODO: look for better solution.
+                    // The only way to show the message to the user (as I know) -
+                    // to throw BadCredentialsException
+                    throw new BadCredentialsException($this->translator->trans('nullAttributeError', [
+                        '%username%' => $username,
+                        '%attribute%' => $attr['ldap_attr']
+                    ]));
+                }
+            }
+
+            $user->enabled = true;
+
+            try {
+                $group = $userService->loadUserGroup($targetGroup);
+            } catch (NotFoundException $exception) {
+                throw new \Exception($this->translator->trans('wrongGroupError', ['%id%' => $targetGroup]));
+            }
+
+            // Create new user under 'admin' user
+            $this->repository->sudo(function (Repository $repository) use ($user, $group, $event, $userService) {
+                try {
+                    $event->setApiUser($userService->loadUserByLogin('admin'));
+                    $userService->createUser($user, [$group]);
+                } catch (NotFoundException $exception) {
+                    $this->tokenStorage->setToken(null);
+                    $event->getRequest()->getSession()->invalidate();
+                    throw new \Exception($this->translator->trans('userNotCreated', ['%username%' => $username]), 0, $exception);
+                } catch (\Exception $exception) {
```

Comment:
Not really important, and might not be relevant here, but as this is php 7.1 and up this could have been using `catch (NotFoundException | Exception $exception)`